extensions/warp-ipfs/src/thumbnail.rs

```diff
@@ -203,23 +206,49 @@ impl ThumbnailGenerator {
 
                             let data = bytes.await.map(|b| b.to_vec())?;
                             let cursor = Cursor::new(data);
-                            let image = ImageReader::new(cursor)
-                                .with_guessed_format()?
-                                .decode()
-                                .map_err(anyhow::Error::from)?;
 
-                            let width = width.min(image.width());
-                            let height = height.min(image.height());
-
-                            let thumbnail = image.thumbnail(width, height);
                             let mut t_buffer = std::io::Cursor::new(Vec::new());
                             let output_format = match (output_exact, format) {
                                 (false, _) => ImageFormat::Jpeg,
                                 (true, format) => format,
                             };
-                            thumbnail
-                                .write_to(&mut t_buffer, output_format)
-                                .map_err(anyhow::Error::from)?;
+                            if output_format == ImageFormat::Gif {
+                                let decoder = GifDecoder::new(cursor).map_err(|_| Error::Other)?;
+                                let frames = decoder
+                                    .into_frames()
+                                    .collect_frames()
+                                    .map_err(|_| Error::Other)?;
+                                let frames = frames.iter().map(|frame| {
+                                    let buffer = frame.buffer().clone();
+                                    let width = width.min(buffer.width());
+                                    let height = height.min(buffer.height());
+                                    let img =
+                                        DynamicImage::ImageRgba8(buffer).thumbnail(width, height);
+                                    Frame::from_parts(
+                                        img.into(),
+                                        frame.left(),
+                                        frame.top(),
+                                        frame.delay(),
+                                    )
+                                });
+                                let mut encoder = GifEncoder::new(&mut t_buffer);
+                                let _ = encoder.set_repeat(Repeat::Infinite);
+                                let _ = encoder.encode_frames(frames);
```

Comment:
We should probably try the result here because if it fails, we want to know so we arent passing a blank buffer into the thumbnail.